src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -113,6 +114,11 @@ public AWSCredentials getCredentials() {
         if (StringUtils.isBlank(iamRoleArn)) {
             return initialCredentials;
         } else {
+            // Handle the case of delegation to instance profile
+            if (StringUtils.isBlank(accessKey) && StringUtils.isBlank(secretKey.getPlainText()) ) {
+                initialCredentials = (new InstanceProfileCredentialsProvider()).getCredentials();
```

Comment:
I know this has already been merged and released, but isn't the manually-specified IAM role in the credentials entry (`iamRoleArn`) redundant in this case? The instance profile already contains an assigned IAM role, which is what should be used in this case, no?

Reply:
The use-case for this feature is to specifically switch to a different account's role.
Useful in cross-account integrations.

Instance Profile is needed for this (unless creds are used) to authenticate that you are allowed to do this switch, and role is needed as a target.